frontend/src/metabase-lib/lib/queries/StructuredQuery.ts

```diff
@@ -922,7 +946,8 @@ class StructuredQueryInner extends AtomicQuery {
     return this.table().segments.filter(
       segment =>
         (currentSegmentId != null && currentSegmentId === segment.id) ||
-        (!segment.archived && !this.segments().includes(segment)),
+        (!segment.archived &&
+          (includeAppliedSegments || !this.segments().includes(segment))),
```

Comment:
The existing logic was to filter out applied segments from the list of filter options. This didn't make sense for the dropdown UI. This adds an option to bypass that behavior for the filter modal, while preserving the existing behavior elsewhere.